pkg/chunked/storage_linux.go

```diff
@@ -146,40 +146,48 @@ func (c *chunkedDiffer) convertTarToZstdChunked(destDirectory string, payload *o
 func GetDiffer(ctx context.Context, store storage.Store, blobDigest digest.Digest, blobSize int64, annotations map[string]string, iss ImageSourceSeekable) (graphdriver.Differ, error) {
 	pullOptions := store.PullOptions()
 
+	convertImages := parseBooleanPullOption(pullOptions, "convert_images", false)
+
 	if !parseBooleanPullOption(pullOptions, "enable_partial_images", true) {
-		return nil, errors.New("enable_partial_images not configured")
+		return nil, internal.ErrPartialContentNotAvailable
 	}
 
 	zstdChunkedTOCDigestString, hasZstdChunkedTOC := annotations[internal.ManifestChecksumKey]
 	estargzTOCDigestString, hasEstargzTOC := annotations[estargz.TOCJSONDigestAnnotation]
 
 	if hasZstdChunkedTOC && hasEstargzTOC {
-		return nil, errors.New("both zstd:chunked and eStargz TOC found")
+		if !convertImages {
+			return nil, fmt.Errorf("both zstd:chunked and eStargz TOC found: %w", internal.ErrPartialContentNotAvailable)
+		}
+		// If a conversion is required, ignore both TOCs and retrieve the entire layer.
+		hasZstdChunkedTOC, hasEstargzTOC = false, false
```

Comment:
I’d rather fail without fallback here; this is either a _severe_ misconfiguration or an attack.

(c/image is not ideal in this right now, https://github.com/containers/image/blob/93fa49b0f1fb78470512e0484012ca7ad3c5c804/copy/single.go#L892-L894 , but still, that shouldn’t result in an image with both TOC annotations.)